### DISCUSSION TITLE
Running code on terminal resize after layout is redrawn

### DISCUSSION THREAD

**Discussion by SkwalExe** [Textualize/textual#3527] [Q&A] [answered]
When using the on_resize event the function runs before the layout is redrawn. I need to run code when the terminal is resized but AFTER the layout has been redrawn (when the widgets' size are adjusted etc). Is it possible to do so?

**Comment by davep** [comment]
I don't know for sure, but probably the first place I'd look would be to see if [`call_after_refresh`](https://textual.textualize.io/api/message_pump/#textual.message_pump.MessagePump.call_after_refresh), made from `on_resize`, does the trick.

More generally, perhaps you could explain what it is you're wanting to do or why -- sometimes there might be a better approach to the core problem.

**Reply by SkwalExe** [reply]
I need to get a widget's size after the terminal resize. I dont know how I could implement a call_after_resize function...

**Reply by davep** [reply, marked as answer]
The suggestion was to call `call_after_refresh` from inside a `on_resize`. For example:

```python
from textual.app import App, ComposeResult
from textual.widgets import Static

class WidgetSizeApp(App[None]):

    CSS = """
    #fr1 {
        height: 1fr;
    }
    #fr2 {
        height: 2fr;
    }
    """

    def compose(self) -> ComposeResult:
        yield Static(id="fr1")
        yield Static(id="fr2")

    def on_mount(self) -> None:
        self.call_after_refresh(self.show_sizes)

    def on_resize(self) -> None:
        self.call_after_refresh(self.show_sizes)

    def show_sizes(self) -> None:
        for widget in self.query(Static):
            widget.update(f"{widget.size!r}")

if __name__ == "__main__":
    WidgetSizeApp().run()
```

**Reply by davep** [reply]
Was this what you were after @SkwalExe?

**Reply by SkwalExe** [reply]
Yes, it worked, thank you for your help @davep !
